include/capstone/loongarch.h

```diff
@@ -0,0 +1,239 @@
+#ifndef CAPSTONE_LOONGARCH_H
+#define CAPSTONE_LOONGARCH_H
+
+/* Capstone Disassembly Engine */
+/* By Jiajie Chen <c@jia.je> 2024 */
+/*    Yanglin Xun <1109673069@qq.com>, 2024 */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if !defined(_MSC_VER) || !defined(_KERNEL_MODE)
+#include <stdint.h>
+#endif
+
+#include "platform.h"
+
+//> Operand type for instruction's operands
+typedef enum loongarch_op_type {
+	LOONGARCH_OP_INVALID = 0, // = CS_OP_INVALID (Uninitialized).
+	LOONGARCH_OP_REG, // = CS_OP_REG (Register operand).
+	LOONGARCH_OP_IMM, // = CS_OP_IMM (Immediate operand).
+	LOONGARCH_OP_MEM, // = CS_OP_MEM (Memory operand).
+} loongarch_op_type;
+
+// Instruction's operand referring to memory
+// This is associated with LOONGARCH_OP_MEM operand type above
+typedef struct loongarch_op_mem {
+	unsigned int base;	// base register
+	int64_t disp;	// displacement/offset value
+} loongarch_op_mem;
+
+// Instruction operand
+typedef struct cs_loongarch_op {
+	loongarch_op_type type;	// operand type
+	union {
+		unsigned int reg;	// register value for REG operand
+		int64_t imm;		// immediate value for IMM operand
+		loongarch_op_mem mem;	// base/disp value for MEM operand
+	};
+} cs_loongarch_op;
+
+// Instruction structure
+typedef struct cs_loongarch {
+	// Number of operands of this instruction, 
+	// or 0 when instruction has no operand.
+	uint8_t op_count;
+	cs_loongarch_op operands[8]; // operands for this instruction.
+} cs_loongarch;
+
+//> LOONGARCH registers
+typedef enum loongarch_reg {
```

Comment:
Register names are also generated. Check out the `arm.h` header how it is includedused there.
```
	// generated content <LoongArchGenCSRegEnum.inc> begin
	// clang-format off

    // Here the register enum is inserted by `auto-sync/src/autosync/HeaderPatcher.py`    

	// clang-format on
	// generated content <LoongArchGenCSRegEnum.inc> end
```